hornetq-server/src/main/java/org/hornetq/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -1048,7 +1049,8 @@ public void processRoute(final ServerMessage message, final RoutingContext conte
 
             // We need to kick delivery so the Queues may check for the cursors case they are empty
             schedulePageDelivery(tx, entry);
-            continue;
+
+            paged = true;
```

Comment:
You almost broke paging here...

The only reason you didn't break it.. was because you reverted your own changes down here by testing if (paged)... reference.setPaged(reference) and basically not putting the message anywhere.


The message is delivered through page consumers...



You are doing a big round trip for a very simple change Howard.... no need to change any big semantics on paging...


Simply add a counter for number of paged messages somewhere without changing the whole implementation of PageCounters.